python/cudf/cudf/core/dataframe.py

```diff
@@ -3528,66 +3528,144 @@ def quantile(
     #
     # Stats
     #
+    def _as_common_dtype(self):
+        common_dtype = np.find_common_type(self.dtypes, [])
+        coerced = self._apply_support_method("astype", dtype=common_dtype)
+        return coerced
+
+    def _prepare_for_rowwise_op(self):
+        """Prepare a DataFrame for CuPy-based row-wise operations.
+
+        Returns
+        -------
+        CuPy.NDArray
+        """
+        if not utils.IS_CUPY_AVAILABLE:
+            msg = (
+                "Row-wise operations currently require CuPy. "
+                " Please install CuPy to use these operations."
+            )
+            raise ImportError(msg)
+
+        import cupy as cp
+
+        if any([col.has_null_mask for col in self._columns]):
+            msg = (
+                "Row-wise operations do not currently support columns with "
+                "null values. Consider removing them with .dropna() "
+                "or using .fillna()."
+            )
+            raise ValueError(msg)
+
+        coerced = self._as_common_dtype()
+        arr = cp.asarray(coerced.as_gpu_matrix())
+        return arr
+
+    def _apply_rowwise_op(self, op, numeric_only=None, **kwargs):
+        """Apply a row-wise reduction or scan operation on a DataFrame.
+        """
+
+        # Currently, we don't support row-wise operations on
+        # datetimes, strings, and categoricals.
+        if numeric_only not in (None, True):
+            msg = (
+                "Row-wise operations currently only support int, float, "
+                "and bool dtypes."
+            )
+            raise TypeError(msg)
+
+        filtered = self.select_dtypes(include=[np.number, np.bool])
+        arr = filtered._prepare_for_rowwise_op()
+        result = getattr(arr, op)(axis=1, **kwargs)
+
+        if len(result.shape) == 1:
+            return Series(result, index=self.index)
+        else:
+            result_df = DataFrame.from_gpu_matrix(result).set_index(self.index)
+            result_df.columns = filtered.columns
+            return result_df
+
     def count(self, **kwargs):
         return self._apply_support_method("count", **kwargs)
 
-    def min(self, **kwargs):
+    def min(self, axis=0, numeric_only=None, **kwargs):
+        if axis == 1:
+            return self._apply_rowwise_op("min", numeric_only=numeric_only)
```

Comment:
This should refactored into the `_apply_support_method` function or a new function since this pattern is being repeated across numerous methods.

Reply:
Refactoring this PR today, so should have a clean, reviewable PR soon.